### PR TITLE
View Hazard Feature and Folder Structure Re-organization

### PR DESCRIPTION
**Pull request by BryanCaldeira**

### CODE REVIEW

**Review by whamester — CHANGES_REQUESTED** [review]
General Comments:
- branch naming convention : https://www.figma.com/file/J25BZNGea79onPj9opUUnT/Development-milestone-criteria?type=whiteboard&node-id=12-21244&t=ClGr00UXsw5Yf9pc-0

- the commit message has a typo

- I see that you have a file named components.js, instead of having the components there we could have a folder named components, and inside one file named after the component, e.g. navbar.js, map.js, this will help us keep the components organized and avoid importing code that we are not using.

- inside assets let's add a helpers folder for helper functions like this one: injectComponent, following the same logic, one file per function. naming convention inject-component.helper.js.
And let's add a second params `id` to this function, because we might not want to inject the html to the root but to other element. you can default the second params as root if you like.

File: assets/css/global.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#map {
```
Comment (whamester) [inline assets/css/global.css:1]: I think we have to think about how to keep the styles and the js components together, so we don't have to add the css separately on each feature, in the meantime, let's move this styles to the feature css file view-hazard/style.css

File: assets/js/components.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      <a class="navbar-brand fs-1 fw-bold mx-4" href="#">CampBuddy</a>
+      <div class="input-group">
+    </nav>
+    <div class="mt-4 mx-4">
+      <div class="d-flex gap-1">
+        <div class="form-control">
+          <img src="/assets/img/icons/icon-search.png">
+          <input class="search--input" placeholder="Search here..."/>
+        </div>
+        <button class="btn btn-light filter--btn"><img src="/assets/img/icons/icon-filter.png"></button>
```
Comment (whamester) [inline assets/js/components.js:14]: Is this class needed `filter--btn`?

File: assets/js/map.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    console.log(e);
+  });
+  L.marker([lat, lng], { icon: greenIcon }).addTo(map);
+  L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+    maxZoom: 19,
+    attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+  }).addTo(map);
+}
+
+const getUserLoaction = () => {
```
Comment (whamester) [inline assets/js/map.js:17]: Hey, @ALFAROSO  check this out to get the current position. I think you might use it.

File: pages/view-hazard/index.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="../../assets/css/variables.css">
+  <link rel="stylesheet" href="../../assets/css/global.css">
+  <!-- <script type="module" src="../js/home.js" defer></script> -->
```
Comment (whamester) [inline pages/view-hazard/index.html:8]: remove this commented line

File: pages/view-hazard/index.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  <link rel="manifest" href="../../manifest.json">
+  <link rel="apple-touch-icon" href="../../assets/css/img/icons/icon-96x96.png">
+  <meta name="apple-mobile-web-app-status-bar" content="#000000">
+  <meta name="theme-color" content="#000000">
+  <!-- leaflet JS for maps -->
+  <link rel="stylesheet" href="../../modules/leaflet/leaflet.css">
+  <script src="../../modules/leaflet/leaflet.js"></script>
+  <script type="module" src="./script.js" defer></script>
```
Comment (whamester) [inline pages/view-hazard/index.html:16]: I think these two
```
  <script type="module" src="./script.js" defer></script>
  <link rel="stylesheet" href="./style.css">
```

should be at last. taking priority over the other styles/js

**Review by whamester — CHANGES_REQUESTED** [review]

File: assets/js/map.js
Comment (ReinhardtBotha) [inline assets/js/map.js:1]: just check the spelling of "getUserLoaction". Should be getUserLocation.

File: assets/js/components.js (hunk 2 above)
Comment (BryanCaldeira) [reply assets/js/components.js:14]: Just added for custom styling which I haven't implemented for now... Will remove or keep it in future

**Pull request merged**